### DISCUSSION TITLE
關於GitHub分支(Branch)

### DISCUSSION THREAD

**Discussion by liangyinglly** [1101-GS4719-C/Python-Community-GS4719-C#657] [General]
這幾天與系上同學要共同寫出一個Java的期末專題
我們共享程式碼的方式是使用Github
但我們只創了一個main分支 只要有人修改就需要同步 總是互相影響
詢問家凱老師麼辦後 才知道原來要各自創自己專用的分支再將自己分支的修改導入到主要分支
很感慨知道的太晚了
奉勸大家以後在進行專題分工時記得創建自己的分支 這樣每個提交的歷史記錄都會被儲存
當發生問題時也可以更容易找到發生問題的地方
在此分享大家如何創建分支
https://backlog.com/git-tutorial/tw/stepup/stepup2_2.html

**Comment by HsuChen88** [comment]
我最近也是在做期末
建立了一個儲存庫來共編
但因為只有兩個人沒有開分支
（其實是因為我不太會用）
但謝謝你提供的資訊
希望未來可以再做更大的專案

**Comment by gnsJhenJie** [comment]
開branch的另一個好處是可以在merge branch的時候才處理conflicts, 如果不同的人用同一支branch同時編輯常常會出現conflict需要處理

**Reply by liangyinglly** [reply]
謝謝補充 之後寫專題時應該體會會更深

**Comment by JSHT** [comment]
題外話，之前有人做了一個網站讓大家一起學習 git 指令，覺得對於熟悉 branch 的各項操作蠻有用的。
[learngitbranching](https://learngitbranching.js.org/?locale=zh_TW)

**Reply by sys511613** [reply]
這網站也太有用了吧，感謝分享
之後有空可以用這個網站來學習git branch有關的東西

**Comment by treeann** [comment]
真的做得到嗎 十載ˋ有點難以想像

**Comment by jason79461385** [comment]
之前我只會用fork而已 感謝分享

**Comment by minniewei** [comment]
我怎麼覺得是同系的同胞

**Comment by yenwei12138** [comment]
感謝分享~以後如果要用GitHub做多人報告會使用這技巧

**Comment by Lynn12378** [comment]
感謝分享，branch可以很好的處理衝突的問題
